docs/DeveloperGuide.md

```diff
@@ -2,12 +2,153 @@
+their distinct file paths.
+![storageSave.png](./Diagrams/storageSave.png)
+
+### adding of new entry
+![addCommandSequence.jpg](./Diagrams/addCommandSequence.jpg)
+To add new entries, user will have to input the related commands.
+Wiagi will then parse the command to the AddCommand class.
+The AddCommand class will then validate the user's input and add the input to IncomeList or SpendingList
+
+### deleting of entry, editing of entry
+The commands are similar where there would be a parsing of command to each of its individual classes.
+A similar validation process takes place and actions would be made on IncomeList or SpendingList accordingly
+(deleting entry from list for delete and editing of entry from list for edit)
+
+### Recurrence Component
+
+#### Motivation behind the component:</br>
++ Allows the user to set specific expenditure and incomes as recurring events to increase efficiency when using the
+  application
++ Users may have differing frequencies for recurring events thus application gives them a few common options
+
+Illustrated below is the class diagram for the Recurrence Component:</br>
+</br>
+<img src="./Diagrams/recurrenceClassDiagram.png" alt="recurrenceClassDiagram" width="800"/>
+</br>
+</br>
+Illustrated below is the sequence diagram of the Recurrence Component: </br>
+</br>
+<img src="./Diagrams/recurrenceSequenceDiagram.png" alt="recurrenceSequenceDiagram" width="700"/>
+</br>
+For the reference fragment of 'load from storage', refer to [Storage component](#storage). </br>
+For the reference fragment of 'add recurring entry', refer to 
+[checkIncomeRecurrence / checkSpendingRecurrence](#checkincomerecurrence--checkspendingrecurrence-method) method. </br>
+
+
+#### How the Recurrence Component works:</br>
++ Upon running the application by the user, `Storage` component will load the `IncomeList` and `SpendingList` members of
+`Wiagi` to retrieve past data.
++ Both list are then iterated through. Each member of the list is parsed through `Parser` which returns the type of 
+recurrence it is (eg. `DailyRecurrence`, `null`) which is encapsulated as a `Recurrence` object.
++ If `Recurrence` is not `null` (ie. a recurring entry), it checks the entry and adds to the `SpendingList` and 
+`IncomeList` if needed. </br>
+
+#### Implementation:
+#### Recurrence class
+The `Recurrence` class is a abstract class that provides the interface for checking `Income` and `Spending` and adding 
+recurring entries into the list. </br>
+The following are the abstract methods defined: </br>
++ `checkSpendingRecurrence`
++ `checkIncomeRecurrence`
+
+The following are child classes of `Recurrence`:
++ `DailyRecurrence`: Handles entries labelled as daily recurring events
++ `MonthlyRecurrence`: Handles entries labelled as monthly recurring events
++ `YearlyRecurrence`: Handles entries labelled as yearly recurring events
+
+##### parseRecurrence method
+Class: `Parser` </br>
```

Comment:
What do you mean here? Is it part of the Parser class? Why then is it located under the implementation of the Recurrence class?

Potentially you could help the reader understand the structure a bit more but not only relying on your class diagrams and sequence diagrams, but also by giving you test more structure.

Reply:
I guess what you want to say with the 'class' label is in which classes it is used.

Simply change the naming then to make this more clear.